### ISSUE TITLE
Incorrect typing in query-interface.d.ts for postgres

### ISSUE THREAD

**Issue by akaRem** [sequelize/sequelize#11420]
Sequelize version 5.18.4

Defining trigger using `sequelize.getQueryInterface().createTrigger(...)`
I have error in typings while using correct types expected in js:

`fireOnArray` must be one of insert, update, delete, truncate, while in typings it's declared as object.

here are refs with extracted snippets:

https://github.com/sequelize/sequelize/blob/master/types/lib/query-interface.d.ts#L528
```ts
fireOnArray: {
      [key: string]: unknown;
    }[],
```

https://github.com/sequelize/sequelize/blob/master/lib/dialects/postgres/query-generator.js#L738
```js
    return _.map(fireOnSpec, (fireValue, fireKey) => {
      const EVENT_MAP = {
        'insert': 'INSERT',
        'update': 'UPDATE',
        'delete': 'DELETE',
        'truncate': 'TRUNCATE'
      };

      if (!EVENT_MAP[fireValue]) {
        throw new Error(`parseTriggerEventSpec: undefined trigger event ${fireKey}`);
      }
```

PS: I'd recommend to extend error with `Object.keys(EVENT_MAP)` as available events and `fireValue` which doesn't have corresponding key in `EVENT_MAP`.

**Comment by ephys** [comment]
Yep, looks like it should be `fireOnArray: Array<'UPDATE' | 'INSERT' | 'DELETE' | 'TRUNCATE'>` instead (or is it lowercase?)
